pkg/networkserver/downlink.go

```diff
@@ -688,9 +690,11 @@ func downlinkPathsFromMetadata(
 				},
 			},
 		}
-		if md.PacketBroker != nil {
+		switch {
+		case md.PacketBroker != nil:
+			path.GatewayIdentifiers = packetbroker.GatewayIdentifiers
```

Comment:
The reason to explicitly fill in the gateway identifiers here is that `proto.Equal` shorts to `true` if the pointers of the messages are equal. If this is too much of a micro-optimization, we can just put the `path.GatewayIdentifiers = md.GatewayIds` outside of the switch.

Reply:
It probably matters very little but it looks more consistent also so let's do it this way. It's generally better to write these things in code comments than review comments but no need to change here.